src/coreclr/pal/src/thread/context.cpp

```diff
@@ -1880,3 +1880,29 @@ DBG_FlushInstructionCache(
 #endif
     return TRUE;
 }
+
+#ifdef HOST_AMD64
+CONTEXT& CONTEXT::operator=(const CONTEXT& ctx)
+{
+    size_t copySize;
+    if (ctx.ContextFlags & CONTEXT_XSTATE & CONTEXT_AREA_MASK)
+    {
+        if ((ctx.XStateFeaturesMask & XSTATE_MASK_AVX512) == XSTATE_MASK_AVX512)
+        {
+            copySize = sizeof(CONTEXT);
+        }
+        else
+        {
+            copySize = offsetof(CONTEXT, KMask0);
+        }
+    }
+    else
+    {
+        copySize = offsetof(CONTEXT, Ymm0H);
+    }
+
+    memcpy(this, &ctx, copySize);
```

Comment:
Would it have better performance characteristics to do `memcpy` with a constant in each of the arms above? (Alternatively, copy the first part unconditionally, then the additional parts only conditionally)? I have not looked at codegen, and maybe the C++ compilers already do this optimization, but `memcpy` with a constant length is usually handled much more efficiently by the compilers.

Reply:
I had it that way originally (memcpy in each of the arms), but the compiler actually did compute the size first and then called memcpy once. I liked that, so I've transformed the source to the same way.